### ISSUE TITLE
Improve docs on Interceptors

### ISSUE THREAD

**Issue by tastypackets** [nodejs/undici#3218]
## This would solve...

Currently docs have limited information for interceptors and it's not easy to discover them, Dispatchers.compose() is probably the best documentation in the docs site and it lacks an example or information on the handler callbacks. There is what looks like an outdated [DispatchInterceptor](https://github.com/nodejs/undici/blob/main/docs/docs/api/DispatchInterceptor.md) markdown file in the docs, which is no longer part of the docs sidebar or links. I think this is left over from the interceptors param on Client that's deprecated?

## The implementation should look like...

Add more detailed documentation on interceptors, ideally it should be easier to discover and include handlers examples and or documentation on the handlers.

## Additional context

If this issue is still open for a bit I can try and open a PR to help expand the docs on interceptors, but wanted to open the issue to make sure it's tracked. 🙂

**Comment by metcoder95** [comment]
You are right, the `DispatchInterceptro` is a concept that will be dropped in next major, so currently its usage is not recommended but `Dispatcher.compose` instead.

Will be nice to have an enhancement on the documentation, we are aware we need to support in there.

I believe we can start with an example of an interceptor with custom handlers and link to the handler's documentation, but open to suggestions 🙂 

A PR is always welcomed
